source/MaterialXGenShader/Util.cpp

```diff
@@ -447,4 +447,80 @@ bool isTransparentSurface(ElementPtr element, const ShaderGenerator& shadergen)
     return false;
 }
 
+ValuePtr getImplementationValue(const ValueElementPtr& elem, const InterfaceElementPtr impl, const NodeDef& nodeDef,
```

Comment:
Basically take the value V:
1. See if the nodedef has enums to map to.
2. See if the impl has enums to map to.
3. Find index of V in enums in nodedef.
4. Use that index to find the enum value from the implementation. 
This would be greatly simplified if the value on a enumerated parameter was just the index for instance but is not the case currently. See what Doug says.